### PR TITLE
Use V2 URLs

### PR DESCRIPTION
**Pull request by bergundy**
Implements the V2 spec (for KubeCon).
Should theoretically support pre account URLs when `accountId` is not found to avoid breaking current customers' integrations.
Depends on integration from other services with this branch.

CC: @michaeladda

### CODE REVIEW

File: test/spec.yml [hunk 1]
```diff
@@ -63,6 +63,32 @@
                   bn logs foo --since 13hours
                   bn logs foo --since 9s*
 
+- test: Test old endpoints
+  cleanup:
+    - bn remove momentousgiants
```
Comment (michaeladda) [inline test/spec.yml]: please use a new function name, to avoid confusion.

File: test/spec.yml [hunk 2]
```diff
@@ -63,6 +63,32 @@
...
+              (use "bn create node8 hello" to create a Node.js template function in your CWD)
+    -   in: bn deploy momentousgiants
+        out: |-
+          Deployed function momentousgiants
+          Invoke with one of:
+            "bn invoke momentousgiants"
+            "curl -H X-Binaris-Api-Key:* https://*"
+    -   in: bn invoke momentousgiants
+        out: '"Hello World!"'
+
```
Comment (michaeladda) [inline test/spec.yml]: please check `bn logs` and `bn list` as well for the old endpoints

File: lib/index.js [hunk 3]
```diff
@@ -259,15 +263,30 @@ const loginHandler = async function loginHandler() {
...
+
+  const apiKeyIdx = rawApiKey.lastIndexOf('-');
+  const apiKey = rawApiKey.substr(apiKeyIdx + 1);
+  const realm = rawApiKey.substr(0, apiKeyIdx);
+
+  if (realm) {
+    forceRealm(realm);
   }
-  await updateAPIKey(answer.apiKey);
+  const { accountId, error } = await auth.verifyAPIKey(apiKey);
```
[the remark this answers is not in the document]
Comment (bergundy) [reply lib/index.js]: I wanted to preserve the behaviour of the previous login flow

**Comment by michaeladda** [comment]
please notice that since there are multiple contributors to this PR, i've switched to use merges and not rebases so we do not accidentally force push over each other changes.

**Pull request merged**